skimage/filters/thresholding.py

```diff
@@ -236,16 +236,66 @@ def threshold_local(image, block_size, method='gaussian', offset=0,
     return thresh_image - offset
 
 
-def threshold_otsu(image, nbins=256):
+def _validate_image_histogram(image, hist, nbins=None):
+    """Ensure that either image or hist were given, return valid histogram.
+
+    If hist is given, image is ignored.
+
+    Parameters
+    ----------
+    image : array or None
+        Grayscale image.
+    hist : array, 2-tuple of array, or None
+        Histogram, either a 1D counts array, or an array of counts together
+        with an array of bin centers.
+    nbins : int, optional
+        The number of bins with which to compute the histogram, if `hist` is
+        None.
+
+    Returns
+    -------
+    counts : 1D array of float
```

Comment:
:thinking: Counts are integers... I can see you are casting them into floats line 279. But the `counts` returned by `exposure.histogram` have dtype `int`...

Reply:
@mkcor yes, but basically every calling function was then converting them to float, so I thought that conversion might as well get rolled into here.

Reply:
Ok!